pkg/api/validation/validation.go

```diff
@@ -2348,9 +2348,16 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 		activeDeadlineSeconds := *oldPod.Spec.ActiveDeadlineSeconds
 		mungedPod.Spec.ActiveDeadlineSeconds = &activeDeadlineSeconds
 	}
+
+	// Allow tolerations updates.
+	if len(newPod.Spec.Tolerations) > 0 {
+		allErrs = append(allErrs, validateTolerations(newPod.Spec.Tolerations, fldPath.Child("tolerations"))...)
+	}
+	mungedPod.Spec.Tolerations = oldPod.Spec.Tolerations
```

Comment:
changing/removing tolerations opens the possibility of scheduling a pod based on stale tolerations, and binding it to a tainted node after the pod changed its mind and became intolerant of that taint. only allowing adding tolerations means a previously chosen node would still be tolerated.